src/index.ml

```diff
@@ -543,11 +543,20 @@ struct
     in
     (go [@tailcall]) true 0L 0 0
 
+  let merge_counter =
+    let n = ref 0 in
+    fun () ->
+      incr n;
+      !n
+
   let merge ?(blocking = false) ?(filter = fun _ -> true) ?(hook = fun _ -> ())
       ~witness t =
     let yield () = check_pending_cancel t in
     Semaphore.acquire t.merge_lock;
-    Log.info (fun l -> l "[%s] merge" (Filename.basename t.root));
+    let merge_id = merge_counter () in
+    let merge_start_time = Mtime_clock.counter () in
```

Comment:
there is also this timer for merge https://github.com/mirage/index/blob/master/src/index.ml#L668; maybe it's not worth keeping both? (you can remove the one in stats)

Reply:
I avoided doing that originally since I wanted to keep the statistics about merge times, but it's fairly easy to just pass the merge duration to `Stats` instead. I've pushed a commit that does that :+1:

Reply:
Thanks! merge whenever you're ready